omdb/src/bin/omdb/nexus.rs

```diff
@@ -157,13 +193,285 @@ fn print_task(bgtask: &BackgroundTask) {
                 reason_str(&last.reason)
             );
             print!(
-                "    started at {} and ran for {:.3}ms\n",
+                "    started at {} ({}s ago) and ran for {:.3}ms\n",
                 humantime::format_rfc3339_millis(last.start_time.into()),
+                (Utc::now() - last.start_time).num_seconds(),
                 std::time::Duration::from(last.elapsed.clone()).as_millis(),
             );
         }
     };
 
+    // Print extra task-specific information.  This data is particularly
+    // unstable -- it gets exposed by background tasks as unstructured
+    // (schemaless) data.  We make a best effort to interpret it.
+    if let LastResult::Completed(completed) = &bgtask.last {
+        print_task_details(&bgtask, &completed.details);
+    }
+}
+
+/// Interprets the unstable, schemaless output from each particular background
+/// task and print a human-readable summary
+///
+/// Each activation of a background task produces an arbitrary JSON value that
+/// gets passed to us here.  The task can put whatever status it wants here --
+/// it's solely for debugging.  This function decodes these values and prints
+/// them out.
+///
+/// As implied by not having a schema, the specific values are currently
+/// undocumented and unstable (subject to change).  That does make this code
+/// both ugly and brittle.  It's not a fatal error to fail to parse these, but
+/// we do warn the user if that happens.
+fn print_task_details(bgtask: &BackgroundTask, details: &serde_json::Value) {
+    // All tasks might produce an "error" property.  If we find one, print that
+    // out and stop.
+    #[derive(Deserialize)]
+    struct TaskError {
+        error: String,
+    }
+    if let Ok(found_error) =
+        serde_json::from_value::<TaskError>(details.clone())
+    {
+        println!("    last completion reported error: {}", found_error.error);
+        println!("");
+        return;
+    }
+
+    // The rest is task-specific, keyed by the name.
```

Comment:
For things keyed by name, what if, if they don't deserialize, we instead format the json output rather than return just an error? One of the benefits of json is that it has named/tagged values, so we can just dump the value pretty-printed if we can't manually format it nicely.

If we don't want to clutter our output with pretty-printed json we could provide an option to skip this and just return the error.

Reply:
Yes, great idea.  (I actually thought we were already doing that, but we only do that right now for tasks we don't recognize.)